include/matrix_operations/EigenvaluesProblems.h

```diff
@@ -439,10 +439,38 @@ class EigenvaluesProblems<NT, Eigen::Matrix<NT,Eigen::Dynamic,Eigen::Dynamic>, E
     /// \param[in] B: symmetric matrix
     /// \return The minimum positive eigenvalue and the corresponding eigenvector
     NT minPosLinearEigenvalue_EigenSymSolver(MT const & A, MT const & B, VT &eigvec) const {
+#if defined(SPECTRA_EIGENVALUES_SOLVER)
+	    int matrixDim = A.rows();
+    NT lambdaMinPositive;
+    
+    Spectra::DenseSymMatProd<NT> op(A);
+    Spectra::DenseCholesky<NT> Bop(B);
+    
+    //construct generalized eigen solver object, requesting the smallest eigenvalue
+    Spectra::SymGEigsSolver<NT, Spectra::SMALLEST_ALGE,  Spectra::DenseSymMatProd<NT>, Spectra::DenseCholesky<NT>, Spectra::GEIGS_CHOLESKY>
+        	geigs(&op, &Bop, 1, 15 < matrixDim ? 15 : matrixDim);
+   	 
+    	//initialize and compute
+    	geigs.init();
+    	int nconv = geigs.compute();
+   	 
+    	//retrieve results
+    	VT evalues;
+   	 
+    	if(geigs.info() == Spectra::SUCCESSFUL){
+   		 evalues = geigs.eigenvalues();
+   		 eigvec = geigs.eigenvectors().col(0);
+    	}
+   	 
+    	lambdaMinPositive = evalues(0);
```

Comment:
Since we compute the largest eigenvalue here we need `lambdaMinPositive = NT(1) / evalues(0);` to obtain the smallest positive eigenvalue.